### DISCUSSION TITLE
safelist isn't working for conditional classes hover, focus etc.

### DISCUSSION THREAD

**Discussion by vamshinenu** [tailwindlabs/tailwindcss#12161] [Help]
I am using 
tailwinds : 3.3.3
svelte: 4.0.5

I have a +page.js that will have colors, like `gray-400`, `red-400` and I'm sending them over using `load`,
and in my svelte component I am using it as `text-` and `bg-` and also `hover:bg-` `hover:text-`
I have these classes in the tailwind.config.js under safelist as `hover:bg-gray-400` and also for other colors, and I also have a const possibleColors and set all of them including `hover:bg-gray-400` etc, but its not working that way.

This app has folders/icons, that have different colors and I have trying to change the bg of the actions (like download button, share button) on hover to the color of the folder color.

```const newClass = `hover:bg-${$selectedFile.iconColor.split('-').splice(1).join('-')}`;```

In local: I have to refresh my browser to make it
but my app when deployed to vercel, refresh isn't working.

Not sure if I am missing anything here.

**Comment by maxchehab** [comment]
Have you added the conditional classes to the variants like this?

```ts
safelist: [
    { pattern: /ring-./, variants: ['focus'] },
]
```

More info: https://github.com/tailwindlabs/tailwindcss/discussions/10079#discussioncomment-4402529

**Reply by vamshinenu** [reply]
`	safelist: [
		'text-gray-400',
		'text-emerald-400',
... and more
		'bg-red-400',
		'bg-blue-400',
... and more
		'hover:bg-red-400',
		'hover:bg-gray-400',
... and more
	],`

this is what I have. and Let me try that actually.
the question is that how can I do a pattern 
`{pattern: /bg-./, variants: ['hover'] }` ? (but did not work tho)

**Comment by alex-krasikau** [comment]
Hello @vamshinenu, could you please provide a minimal reproduction?

**Reply by vamshinenu** [reply]
I've tried to recreate this, but it works the way I wanted to. but not in my current app, where the case is the same. 

https://github.com/vamshinenu/hoverbgtest

I have made a similar one in this svelte app. (which works fine even without safe lists)

**Reply by alex-krasikau** [reply]
The reason why it works in the reproduction is that, [in this declaration](https://github.com/vamshinenu/hoverbgtest/blob/main/src/routes/%2Bpage.svelte#L4), Tailwind CSS is able to parse the tokens that you use at runtime and add them to the resulting CSS. So to say, it's safe to list these tokens.

However, until you're able to reproduce the issue, it would be hard to find out the cause on our side.

Another point to consider is that a safe list is usually the worst solution among other options, as stated in the documentation. Maybe if you're able to better explain the goal you're trying to achieve, we will be able to suggest a better solution.

Usually, when you have dynamic colors based on the user input, it's better to use inline custom variables (sorry, the syntax is React, but should be clear):
```js
const color = "#3b82f6";

<div style={{ "--selected-color": color }}>
  <span className="text-[--selected-color]">Content</span>
  <button className="bg-[--selected-color]">Click me</button>
</div>
```
